tests/span_tests.cpp

```diff
@@ -589,10 +589,7 @@ TEST(span_test, first)
 
     {
         span<int, 5> av = arr;
-#ifdef FAIL_ON_SOME_PLATFORMS
-        EXPECT_TRUE(av.first<6>().size() == 6);
-        EXPECT_TRUE(av.first<-1>().size() == -1);
-#endif
+        EXPECT_DEATH(av.first<6>().size(), expected);
```

Comment:
This looks like a regression. I would expect this to be checked at compile time, not at runtime.

Reply:
I agree. `span<T, Extent>::first<N>()` should ensure that `N` and `Extent` play nicely together.

Reply:
That being said - it isn't a PR blocker. If you want to fix it, go ahead. If not, I'm sure another PR will take care of it.